pkg/envconf/config.go

```diff
@@ -31,6 +32,7 @@ import (
 
 // Config represents and environment configuration
 type Config struct {
+	clientLock              sync.RWMutex
```

Comment:
Do you really need an`RWMutex` here ?  Since you are only using the `Mutex` part of the workflow, would it not be enough to use just the `sync.Mutex` part of it ?

Reply:
@harshanarayana You're right. Changed it to `sync.Mutex`